freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -187,10 +194,8 @@ public Host addHost(String fqdn) throws FreeIpaClientException {
     }
 
     public User deleteUser(String userUid) throws FreeIpaClientException {
-        FreeIpaChecks.checkUserNotProtected(userUid, () -> String.format("User '%s' is protected and cannot be deleted from FreeIPA", userUid));
-        List<Object> flags = List.of(userUid);
-        Map<String, Object> params = Map.of();
-        return (User) invoke("user_del", flags, params, User.class).getResult();
+        return UserRemoveOperation.get(userUid).invoke(this).orElseThrow(() ->
```

Comment:
I think there are several places that might be looking for `FreeIpaErrorCodes.NOT_FOUND` in the original exception and I am concerned that this might not be handled correctly in the bulk case. I don't have something specific to point to but I wanted to check if this case was considered.

Reply:
yes, right, i'll introduce a method to get acceptable error codes and handle it in batch call, in case of remove it will be not_found

Reply:
added it, anyway i tested batch call with useradd operation for user who is existed in freeipa alrady, it seemed batch call handled it, and doesn't even thrown a duplicateeentry error